index.bs

```diff
@@ -239,28 +241,43 @@ following steps:
     2. If |name| exists as a key in the <a>animator name to animator definition map</a>,
         <a>throw</a> a <a>NotSupportedError</a> and abort all these steps.
 
-    3. If the result of <a>IsConstructor</a>(|animatorCtor|) is false, <a>throw</a> a
+    3. If the result of <a>IsConstructor</a>(|animatorCtorValue|) is false, <a>throw</a> a
         <a>TypeError</a> and abort all these steps.
 
-    4. Let |prototype| be the result of <a>Get</a>(|animatorCtor|, "prototype").
+    4. Let |animatorCtor| be the result of <a>converting</a> animatorCtorValue to the
+        <a>VoidFunction</a> <a>callback function</a> type.  Rethrow any exceptions from the
+        conversion and abort all these steps.
+
+    4. Let |prototype| be the result of <a>Get</a>(|animatorCtorValue|, "prototype").
 
     5. If the result of <a>Type</a>(|prototype|) is not Object, <a>throw</a> a <a>TypeError</a>
         and abort all these steps.
 
-    6. Let |animate| be the result of <a>Get</a>(|prototype|, "animate").
+    6. Let |animateValue| be the result of <a>Get</a>(|prototype|, "animate").
 
-    7. If the result of <a>IsCallable</a>(|animate|) is false, <a>throw</a> a <a>TypeError</a> and
+    7. Let |animate| be the result of <a>converting</a> |animateValue| to the <a>Function</a>
+        <a>callback function</a> type. Rethrow any exceptions from the conversion and
         abort all these steps.
 
-    10. Let |definition| be a new <a>animator definition</a> with:
+    8. Let |destroyValue| be the result of <a>Get</a>(|prototype|, "onDestroy").
+
+    9. Let |destroy| be the result of <a>converting</a> |destroyValue| to the <a>Function</a>
+        <a>callback function</a> type. Rethrow any exceptions from the conversion and
```

Comment:
Ditto; the abort needs to be conditional.

Reply:
Done.